calicoctl/commands/node/status.go

```diff
@@ -48,49 +46,56 @@ Description:
 	// printed for this option
 	_, _ = docopt.Parse(doc, nil, true, "calicoctl", false, false)
 
-	ctx := context.Background()
-
-	if os.Getenv("DOCKER_API_VERSION") == "" {
-		err := os.Setenv("DOCKER_API_VERSION", "1.16")
-		if err != nil {
-			log.Fatalf("Error setting DOCKER_API_VERSION: %v", err)
-		}
-	}
-
-	dockerClient, err := client.NewEnvClient()
-	if err != nil {
-		return err
-	}
-
-	options := types.ContainerListOptions{All: true}
-	containers, err := dockerClient.ContainerList(ctx, options)
+	processes, err := gops.Processes()
 	if err != nil {
-		log.Fatalf("Error getting the container list: %v", err)
+		fmt.Println(err)
 	}
 
-	for _, c := range containers {
-		if strings.Contains(c.Names[0], "calico-node") && c.State == "running" {
-
-			fmt.Printf("calico-node container is running. Status: %s\n", c.Status)
+	// Go through running processes and check if `calico-felix` and `bird` (`bird6`)
+	// processes are running
+	if psContains("calico-felix", processes) {
+		fmt.Printf("Calico process is running.\n")
 
-			if os.Getuid() != 0 {
-				fmt.Println("This command must be run as root.")
-				os.Exit(1)
-			}
+		// Must run this command as root to be able to connect to BIRD sockets
+		if os.Getuid() != 0 {
+			fmt.Println("This command must be run as root.")
+			os.Exit(1)
+		}
 
+		// Check if birdv4 process is running, print the BGP peer table if it is, else print a warning
+		if psContains("bird", processes) {
 			printBGPPeers("4")
-			printBGPPeers("6")
+		} else {
+			fmt.Printf("\nINFO: BIRDv4 process: 'bird' is not running.\n")
+		}
 
-			fmt.Println()
-			return nil
+		// Check if birdv6 process is running, print the BGP peer table if it is, else print a warning
+		if psContains("bird6", processes) {
+			printBGPPeers("6")
+		} else {
+			fmt.Printf("\nINFO: BIRDv6 process: 'bird6' is not running.\n")
 		}
+
+		// Have to manually enter an empty line because the table print
+		// library prints the last line, so can't insert a '\n' there
+		fmt.Println()
+		return nil
 	}
 
 	// Return and print message if calico-node is not running
-	fmt.Println("calico-node container not running")
+	fmt.Printf("Calico process is not running.\n")
```

Comment:
I think we should probably be OS exiting in this case: status is definitely not good, so RC 1 seems best.

It would also allow you to refactor the function to make the conditional / return structure simpler: a bunch of unnested if tests, with a couple of exit(1)s for failures, and a single return nil "OK" from the end.